### ISSUE TITLE
Redesign of request page

### ISSUE THREAD

**Issue by bdaley** [GreenhouseStudios/sourcery-web-app#218]
See: https://github.com/GreenhouseStudios/sourcery-web-app/discussions/215

**Comment by briankelleher** [comment]
@bdaley Quick question.

In this function: https://github.com/GreenhouseStudios/sourcery-web-app/blob/master/store/create.js#L155

There is an automatic 'picked up' status if the request is detected to be for a partner repository - is this intended or just put in for demo purposes?  Was this the initial thinking when both an institutional version and p2p version needed to co-exist?  It might be helpful to have some background knowledge on what the 'pickup' statuses are (sorry, should have asked this in meetings prior).

**Comment by bdaley** [comment]
Oh man. That code is a mess.

TLDR - I think you understand the situation correctly.

First off, (and for the record) I don't think this is the correct use of a Vue store. This was also from when we were doing the P2P model and we were all still learning Nuxt. However, we were using the Vue Store system to act as sort of an active record class/pattern. Like, CRUD. This is part of the reason that I started writing separate models recently. So, feel free to get things working with the current code, but I see this disappearing altogether over the summer.

The status setting is complicated, but I'll try to explain with some context...

### Workflow 1 - The P2P Model

1.  **Pending**: Request submitted
2.  **Picked Up**: An independent Sourcerer claimed the job
3.  **Complete**: Job was fulfilled
4.  **Archived**: Removed from the dashboard

### Workflow 2 - The institutional & P2P model

1.  **Reserved**: Job is shown only to the selected/intended institution. The institution can claim it (changes to "picked up") or release it (changed to "pending") and it goes to the open P2P market.
2.  **Pending**
3.  **Picked Up**
4.  **Complete**
5.  **Archived**

### Workflow 3 - Institutional MVP 👈 (you're here)

However, after talking with some institutions we decided to scrap the "Reserved" option for now and just have requests go directly to the institution. So, currently, there's no "pending" or "reserved" status option.

1.  **Picked Up**: The institution has the job in its dashboard to fulfill
2.  **Complete**
3.  **Archived**

For now, Workflow 3 is what we're going with. Long-term, Workflow 2 (or similar) is probably where we're headed.

**Comment by bdaley** [comment]
One thing I forgot to mention -- our cloud functions depend on these changes in "state" to send transactional communications (currently just email).

For example, if the state of a request changes from "picked\_up" to "complete" an email is triggered to let the researcher know that their documents are ready. 

Just something to keep in mind when working with the status field.

**Comment by briankelleher** [comment]
Thank you for all of that background - if we are pursuing Workflow 3 perhaps I can clean up the unnecessary parts and then make sure cloud functions are either updated or not affected in the process.

**Comment by bdaley** [comment]
:+1:

**Comment by foundhistory** [comment]
Thanks, guys, for all your hard work sharing info and getting the team up to speed. (Also, Brian’s description looks righton to me)
